src/sipi.cpp

```diff
@@ -564,11 +619,32 @@ int main(int argc, char *argv[]) {
             try {
                 format = std::string(options[FORMAT].arg);
             } catch (std::exception &err) {
-                std::cout << "##" << __LINE__ << std::endl;
                 std::cerr << options[FORMAT].desc->help;
                 return EXIT_FAILURE;
             }
         }
+        else {
+            //
+            // there is no format option given – we try to determine the format
+            // from the output name extension
+            //
+            size_t pos = outfname.rfind(".");
+            if (pos != std::string::npos) {
+                std::string ext = outfname.substr(pos + 1);
+                if ((ext == "jpx") || (ext == "jp2")) {
```

Comment:
Solved. If the file extension is missing or unknown (and no --format clause is given), an error message is emitted and the command line conversion terminates.